extension/src/repository/errorDecorationProvider.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  Event,
+  EventEmitter,
+  FileDecoration,
+  FileDecorationProvider,
+  ThemeColor,
+  Uri,
+  window
+} from 'vscode'
+import { Disposable } from '../class/dispose'
+
+export const getDecoratableUri = (label: string): Uri =>
+  Uri.from({ path: label, scheme: 'dvc.tracked' })
+
+export class ErrorDecorationProvider
+  extends Disposable
+  implements FileDecorationProvider
```

Comment:
[F] There is a base class here but I'll not split it out until after this PR.

Reply:
It also makes sense (to me at least) to keep this separate from the SCM decoration provider rather than trying to mangle them together.